plugins/inputs/x509_cert/x509_cert_test.go

```diff
@@ -260,6 +262,37 @@ func TestGatherChain(t *testing.T) {
 	}
 }
 
+func TestGatherUDPCert(t *testing.T) {
+	pair, err := tls.X509KeyPair([]byte(pki.ReadServerCert()), []byte(pki.ReadServerKey()))
+	require.NoError(t, err)
+
+	cfg := &dtls.Config{
+		Certificates: []tls.Certificate{pair},
+	}
+
+	addr := &net.UDPAddr{IP: net.ParseIP("127.0.0.1"), Port: 0}
+	listener, err := dtls.Listen("udp", addr, cfg)
+	require.NoError(t, err)
+	defer listener.Close()
+
+	go func() {
+		_, err := listener.Accept()
+		require.NoError(t, err)
```

Comment:
You shall not use any `require` methods that terminate (i.e. call `require.Fail` internally) in goroutines as this will only end the goroutine and not the test. See [the go release notes](https://tip.golang.org/doc/go1.16#vet) for reference also describing a potential workaround.

In this specific instance I think just ignoring the error and quitting the goroutine might be sufficient as the test relying on the accept will probably fail anyway...